src/Components/Web.JS/src/Rendering/BrowserRenderer.ts

```diff
@@ -257,7 +257,6 @@ export class BrowserRenderer {
     if (newDomElementRaw instanceof HTMLSelectElement && selectValuePropname in newDomElementRaw) {
       const selectValue = newDomElementRaw[selectValuePropname];
       newDomElementRaw.value = selectValue;
-      delete newDomElementRaw[selectValuePropname];
```

Comment:
It's nice that removing this fixes the issue. Any idea why this line existed to start with?

Reply:
*summons @SteveSandersonMS*

Reply:
https://github.com/dotnet/aspnetcore/issues/21453#issuecomment-623384973

Reply:
> Any idea why this line existed to start with?

I was trying to avoid cases where state hangs around longer than it's needed and potentially interferes in the future. I didn't know any specific case where it would make a different; just trying to keep things tidy so there is less for us to have to reason about.

But now it's clear there's a definite scenario where it's positively advantageous for the state to hang around until we later find an `<option>` to match it up with.